src/tree-sitter/csharpLanguage.ts

```diff
@@ -0,0 +1,26 @@
+import { TreeSitterLanguage } from './types.js'
+
+export const csharpLanguage: TreeSitterLanguage = {
+  defineParameterTypeQueries: [],
+  defineStepDefinitionQueries: [
+    `
+(method_declaration (attribute_list
+    (attribute
+      name: (identifier) @annotation-name
+      (attribute_argument_list
+        (attribute_argument
+          (verbatim_string_literal) @expression
+        )
+      )
+    )
+  )
+  (#match? @annotation-name "Given|When|Then|And|But|StepDefinition")
+)
+`,
+  ],
+  toStringOrRegExp(s: string): string | RegExp {
+    const match = s.match(/^([@$'"]+)(.*)"$/)
+    if (!match) throw new Error(`Could not match ${s}`)
+    return new RegExp(match[2])
```

Comment:
Not sure if returning regexp for all @expressions is the right thing, but as any of the specflow expressions could be a regex (i.e. [Given(@"Some text followed by regex (\d\.*\d) and another (.*)") I figured this is better than trying to pick out the ones that don't and returning them as plain strings.

Reply:
@Issafalcon I don't know if it helps with the decision, but the Cucumber Expression support (although not integrated yet to core SpecFlow) can be used using the SpecFlow plugin: [CucumberExpressions.SpecFlow](https://www.nuget.org/packages/CucumberExpressions.SpecFlow.3-9). Currently it works in a way that it stranscribes the Cucumber Expression to Regex. The only tricky point is how you can decide if an expression is a cucumber exprssion or regex, which can be done by some heuristics, like [this](https://github.com/gasparnagy/CucumberExpressions.SpecFlow/blob/master/CucumberExpressions.SpecFlow.SpecFlowPlugin/Plugin/StepDefinitionMatchExpressionConverter.cs#L23).

Reply:
@gasparnagy the reason we parse (custom) parameter types is to be able to compile Cucumber Expressions that use them.


A custom [StepArgumentTransformation](https://github.com/gasparnagy/CucumberExpressions.SpecFlow#features) can be used to define custom transformations:

```csharp
[StepArgumentTransformation]
public Person ConvertPerson(string name)
{
  //...
}
```

This transformation is chosen when a step definition has a `Person` parameter in the method definition. However, I am not sure it can be referred to as `{Person}` in a Cucumber Expression (or can it @gasparnagy)? If that's the case, then would it be possible to attach a regex to the annotation?

```csharp
[StepArgumentTransformation]
[ParameterTypeRegExp("[A-Z][a-z]*")]
[ParameterTypeName("person")] // Use as {person}
public Person ConvertPerson(string name)
{
  //...
}
```

I think an API like this could bring Cucumber Expressions for .NET a bit closer to the other implementations. It would also make it easier to build a more powerful auto-complete for LSP clients.

Let me know what you think.

Reply:
@aslakhellesoy Sorry, I wasn't checking that we are talking about the parameter conversions here. 
For your questions:

- If you specify a `[StepArgumentTransformation]` for a `Person` type, you CAN refer to it as `{Person}`, so that works. (You can also refer to it with the jolly joker `{}` notation.)
- You CAN also specify a regex using the same attribute: `[StepArgumentTransformation("([A-Z][a-z]*)")]`. 
- The only difference is that the value you would like to pass in to the method parameter has to be in a regex group. But you can also do things, like `[StepArgumentTransformation("$(\d+)")]` that will match to `$100`, but only the 100 will be passed in to the conversion method (can even have an int parameter). 
- What DOES NOT work yet is to give a name to the parameter type to be used as `{person}`. This is probably will be implemented in a way that you will need to provide an additional name parameter to the attribute: `[StepArgumentTransformation("([A-Z][a-z]*)", Name = "person")]`

Because the `StepArgumentTransformation` attribute already supports specifying the regex (and to be sync with the other configuration options in SpecFlow), I think sticking to the a single attribute is better instead of introducing a new `ParameterTypeName` one.

Happy to continue the discussion on the CukenSpace BTW.